scp/cc/cpio/client_providers/private_key_fetcher_provider/src/azure/azure_private_key_fetcher_provider.cc

```diff
@@ -0,0 +1,198 @@
+/*
+ * Portions Copyright (c) Microsoft Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "azure_private_key_fetcher_provider.h"
+#include "azure_private_key_fetcher_provider_utils.h"
+
+#include <utility>
+#include <vector>
+
+#include "absl/strings/str_cat.h"
+#include "azure/attestation/json_attestation_report.h"
+#include "core/interface/http_client_interface.h"
+#include "cpio/client_providers/interface/auth_token_provider_interface.h"
+#include "cpio/client_providers/interface/role_credentials_provider_interface.h"
+#include "cpio/client_providers/private_key_fetcher_provider/src/private_key_fetcher_provider_utils.h"
+
+#include "error_codes.h"
+
+using google::scp::core::AsyncContext;
+using google::scp::core::ExecutionResult;
+using google::scp::core::FailureExecutionResult;
+using google::scp::core::HttpClientInterface;
+using google::scp::core::HttpHeaders;
+using google::scp::core::HttpRequest;
+using google::scp::core::HttpResponse;
+using google::scp::core::SuccessExecutionResult;
+using google::scp::core::common::kZeroUuid;
+using google::scp::core::HttpMethod;
+using google::scp::core::errors::
+    SC_AZURE_PRIVATE_KEY_FETCHER_PROVIDER_CREDENTIALS_PROVIDER_NOT_FOUND;
+using std::bind;
+using std::placeholders::_1;
+
+namespace {
+constexpr char kAzurePrivateKeyFetcherProvider[] = "AzurePrivateKeyFetcherProvider";
+constexpr char kAuthorizationHeaderKey[] = "Authorization";
+constexpr char kBearerTokenPrefix[] = "Bearer ";
+}  // namespace
+
+namespace google::scp::cpio::client_providers {
+
+ExecutionResult AzurePrivateKeyFetcherProvider::Init() noexcept {
+  RETURN_IF_FAILURE(PrivateKeyFetcherProvider::Init());
+
+  if (!auth_token_provider_) {
+    auto execution_result = FailureExecutionResult(
+        SC_AZURE_PRIVATE_KEY_FETCHER_PROVIDER_CREDENTIALS_PROVIDER_NOT_FOUND);
+    SCP_ERROR(kAzurePrivateKeyFetcherProvider, kZeroUuid, execution_result,
+              "Failed to get credentials provider.");
+    return execution_result;
+  }
+
+  return SuccessExecutionResult();
+}
+
+ExecutionResult AzurePrivateKeyFetcherProvider::SignHttpRequest(
```

Comment:
**[Comment for actual PR]**
We are not doing anything here, because `auth_token_provider` is not implemented yet.